### PR TITLE
Fix stackmap always marking ConstantDynamic as objects

### PR DESCRIPTION
**Pull request by fengxue-IS**
- Check Condy return type before marking stack slot

Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

### CODE REVIEW

File: runtime/stackmap/stackmap.c [hunk 1]
```diff
@@ -350,7 +350,14 @@ mapStack(UDATA *scratch, UDATA totalStack, U_8 * map, J9ROMClass * romClass, J9R
 				}
 
 				if (pool[index].slot2) {
-					PUSH(OBJ);
+					/* Check for primitive values resolved from ConstantDynamic */
+					if (((pool[index].slot2 & J9DescriptionCpTypeMask) == J9DescriptionCpTypeConstantDynamic)
+					&& (pool[index].slot2 >> J9DescriptionReturnTypeShift)
```
Comment (gacholio) [inline runtime/stackmap/stackmap.c]: Do not use implicit zero comparisons. Also, strictly speaking the constant should be on the left of the comparison.

Comment (fengxue-IS) [reply runtime/stackmap/stackmap.c]: fixed

**Comment by gacholio** [comment]
I'd like @DanHeidinga to have a look before merge.

**Comment by DanHeidinga** [comment]
@fengxue-IS It would be good to add tests for this case or identify a JTREG that already covers it

**Comment by gacholio** [comment]
This issue will be difficult to reproduce with random tests - the condy value needs to be on the pending stack when a GC occurs.

Something like:

`func(condyValue, gc())`

where the `gc()` method forces some GCs before returning.

**Comment by gacholio** [comment]
Should be easy enough to craft the above test and see that it crashes every time without the fix from this PR.

**Comment by fengxue-IS** [comment]
I will look into adding a test case for this

**Comment by gacholio** [comment]
@DanHeidinga Does the verifier need a similar change, or is it already there?

**Comment by DanHeidinga** [comment]
> @DanHeidinga Does the verifier need a similar change, or is it already there?

Verifier already does it correctly - see rtverify.c:
https://github.com/eclipse/openj9/blob/10d5686ffabe19806fc71bd3ea42ddbe9e99fdae/runtime/bcverify/rtverify.c#L720-L726

which pushes the ldc type using:
https://github.com/eclipse/openj9/blob/10d5686ffabe19806fc71bd3ea42ddbe9e99fdae/runtime/bcverify/vrfyhelp.c#L420-L425

**Comment by DanHeidinga** [comment]
Jenkins test sanity zlinux jdk11

**Comment by fengxue-IS** [comment]
@DanHeidinga can you please restart the PR build, looks like it failed due to network issue

> [2020-03-06T18:26:29.903Z] Cannot contact rh7-390-4: java.lang.InterruptedException
[2020-03-06T18:35:50.499Z] wrapper script does not seem to be touching the log file in /home/jenkins/workspace/Build_JDK11_s390x_linux_Personal@tmp/durable-c2ff9402
[2020-03-06T18:35:50.499Z] (JENKINS-48300: if on an extremely laggy filesystem, consider -Dorg.jenkinsci.plugins.durabletask.BourneShellScript.HEARTBEAT_CHECK_INTERVAL=86400)

**Comment by DanHeidinga** [comment]
Jenkins test sanity zlinux jdk11

**Comment by DanHeidinga** [comment]
Jenkins test sanity zlinux jdk11

**Comment by DanHeidinga** [comment]
ported to 0.19 release in https://github.com/eclipse/openj9/pull/877

**Pull request merged**